packages/data-stores/src/domain-suggestions/resolvers.ts

```diff
@@ -68,20 +80,40 @@ export function* __internalGetDomainSuggestions( queryObject: DomainSuggestionQu
 		);
 	}
 
-	if ( ! suggestions || suggestions === '' ) {
+	if ( ! suggestions ) {
 		// Other internal server errors
 		return receiveDomainSuggestionsError(
 			translate( 'Invalid response from the server' ) as string
 		);
 	}
 
-	const processedSuggestions = suggestions.map( ( suggestion: DomainSuggestion ) => ( {
-		...suggestion,
-		...( suggestion.raw_price &&
-			suggestion.currency_code && {
-				cost: getFormattedPrice( suggestion.raw_price, suggestion.currency_code ),
-			} ),
-	} ) );
+	// if the query is a FQDN and the results don't have it,
+	// this implies that the user is searching for an unavailable domain name
+	// TODO: query the availability endpoint to find the exact reason why it's unavailable
+	// all the possible responses can be found here https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/domains/registration/availability-messages.js#L40-L390
+	if ( suggestionsLackThisFQDN( suggestions, queryObject.query ) ) {
```

Comment:
When the user searches for an unavailable domain, we mock a DomainSuggestion and prepend it to the list of suggestions.